crowbar_framework/app/models/service_object.rb

```diff
@@ -1186,6 +1189,19 @@ def apply_role(role, inst, in_queue)
           tmprole = RoleObject.find_role_by_name "#{role_name}_remove"
           unless tmprole.nil?
             elem_remove = tmprole.name
+            # save remove intention in #{@bc_name}-databag
+            if databag["deployment"][@bc_name]["elements"]["to_remove"].nil?
+              databag["deployment"][@bc_name]["elements"]["to_remove"] = {}
+            end
+
+            old_nodes.each do |old_node|
+              @logger.debug "saving #{elem_remove} intention for #{old_node}"
+              if databag["deployment"][@bc_name]["elements"]["to_remove"][old_node].nil?
```

Comment:
Maybe use a temporary variable to avoid repetition?

    to_remove = databag["deployment"][@bc_name]["elements"]["to_remove"]
    to_remove[old_node] ||= []
    to_remove[old_node] << elem_remove